doc/source/reference/migration_guide_from_1.x_to_2.0.rst

```diff
@@ -0,0 +1,117 @@
+This is a guide for people moving from NetworkX 1.X to NetworkX 2.0
+
+As you know we have made some major changes to the methods in the Multi/Di/Graph classes.
+Methods that used to return lists are removed and the methods that returned iterators
+are renamed(removing the _iter suffix).
+For example we hade two methods to get nodes of a graph, ``G.nodes()`` and ``G.nodes_iter()``. ``G.nodes_iter()`` is renamed to ``G.nodes()``, hence 
+``G.nodes()`` returns an iterator now instead of list.
+
+The methods changed are explained with examples below.
+
+	>>> import networkx as nx
+	>>> G = nx.complete_graph(5) 
+	>>> G.nodes()
+	<dictionary-keyiterator at 0x102ae8730>
+
+
+People suprised by this output and expecting something like this
+
+	>>> G.nodes()
+	[0, 1, 2, 3, 4]
+
+don't panic. Nothing is wrong.
+
+Now ``G.nodes()`` returns an iterator. If you don't like this you can always get the old output by
+
+	>>> list(G.nodes())
+	[0, 1, 2, 3, 4]
+
+In the same way for ``G.edges()``
+
+	>>> G.edges()
+	<generator object edges at 0x102ae9fa0>
+	>>> list(G.edges())
+	[(0, 1),
+	 (0, 2),
+	 (0, 3),
+	 (0, 4),
+	 (1, 2),
+	 (1, 3),
+	 (1, 4),
+	 (2, 3),
+	 (2, 4),
+	 (3, 4)]
+
+and ``G.neighbors(n)`` where n is a node.
+ 
+	>>> G.neighbors(2)
+	<dictionary-keyiterator at 0x102ae8aa0>
+
+	>>> list(G.neighbors(2))
+	[0, 1, 3, 4]
+
+So, basically you can switch back to the old behavior by adding `list()` to the method.
+
+BUT we did something different with ``G.degree()``. If a single node is passed as argument,
```

Comment:
For methods that used to return a dict, like G.degree(), the new iterator version returns a (key,value) 2-tuple so that passing it to Python's dict function recovers the original behavior if desired. Note that passing a single node to the degree method still returns the degree of that node as it always has.